tests-hivemq-platform-operator/src/integrationTest/java/com/hivemq/helmcharts/AbstractHelmChartIT.java

```diff
@@ -1,92 +1,178 @@
+        K8sUtil.waitForHiveMQPlatformStateRunning(client, platformNamespace, PLATFORM_RELEASE_NAME);
+    }
+
+    /**
+     * Override with {@code return false;} to prevent the creation of the Platform Operator namespace in the
+     * {@code @BeforeEach} method.
+     */
+    protected boolean createOperatorNamespace() {
         return true;
     }
 
-    protected boolean cleanupPlatformChart() {
+    /**
+     * Override with {@code return false;} to prevent the creation of the Platform namespace in the
+     * {@code @BeforeEach} method.
+     */
+    protected boolean createPlatformNamespace() {
         return true;
     }
 
-    protected boolean cleanupNamespace() {
+    /**
+     * Override with {@code return false;} to prevent the installation of the Platform Operator chart in the
+     * {@code @BeforeEach} method.
+     */
+    protected boolean installOperatorChart() {
         return true;
     }
 
-    protected void installChartsAndWaitForPlatformRunning(final @NotNull String valuesResourceFile) throws Exception {
-        installChartsAndWaitForPlatformRunning("-f", valuesResourceFile, "--namespace", namespace);
+    /**
+     * Override with {@code return false;} to prevent the uninstallation of the Platform Operator chart in the
+     * {@code @AfterEach} method.
+     */
+    protected boolean uninstallOperatorChart() {
+        return true;
     }
 
-    protected void installChartsAndWaitForPlatformRunning(
-            final @NotNull String valuesResourceFile, final @NotNull String setFileOverride) throws Exception {
-        installChartsAndWaitForPlatformRunning("--set-file",
-                setFileOverride,
-                "-f",
-                valuesResourceFile,
-                "--namespace",
-                namespace);
+    /**
+     * Override with {@code return false;} to prevent the uninstallation of the Platform chart in the
+     * {@code @AfterEach} method.
+     */
+    protected boolean uninstallPlatformChart() {
+        return true;
     }
 
-    protected void installChartsAndWaitForPlatformRunning(
-            final @NotNull String... commands) throws Exception {
-        helmChartContainer.installOperatorChart(OPERATOR_RELEASE_NAME);
-        helmChartContainer.installPlatformChart(PLATFORM_RELEASE_NAME, commands);
+    protected final @NotNull CompletableFuture<String> waitForOperatorLog(final @NotNull String log) {
+        return logWaiter.waitFor(OPERATOR_LOG_WAITER_PREFIX, log);
+    }
 
-        K8sUtil.waitForHiveMQPlatformStateRunning(client, namespace, PLATFORM_RELEASE_NAME);
+    protected final @NotNull CompletableFuture<String> waitForPlatformLog(final @NotNull String log) {
+        return logWaiter.waitFor(PLATFORM_LOG_WAITER_PREFIX, log);
+    }
+
+    protected final @NotNull CompletableFuture<String> waitForInitAppLog(final @NotNull String log) {
+        return logWaiter.waitFor(PLATFORM_LOG_WAITER_PREFIX, ".*\\[HiveMQ Platform Operator Init App\\] " + log);
```

Comment:
Note: Eventually we'll have to add the version number (depending on the merge order in a new PR for the other feature or directly in here).

Reply:
OK - I've just added in here already. We can merge the other [one](https://github.com/hivemq/hivemq-platform-operator/pull/175) first

Reply:
Some ITs are failing due to the expecting new log format as per the [PR](https://github.com/hivemq/hivemq-platform-operator/pull/175). So waiting for that to be merged first.